### ISSUE TITLE
Indentation warnings when the indentation is correct?

### ISSUE THREAD

**Issue by SturmB** [eslint/eslint#7204]
**Tell us about your environment**

* **ESLint Version:** 1.0.6
* **Node Version:** Unknown. Perhaps installed with VS Code?
* **npm Version:** Unknown. Perhaps installed with VS Code?

**What parser (default, Babel-ESLint, etc.) are you using?** Default.

**Please show your full configuration:** Assuming you mean the .eslintrc.* files, here they are. (Seems indenting isn't working here):

*.eslintrc.js:*

```js
module.exports = {
    "env": {
    },
    "extends": "eslint:recommended",
    "parserOptions": {
        "ecmaFeatures": {
            "jsx": true
        }
    },
    "rules": {
        "indent": [
            "error",
            "tab"
        ],
        "linebreak-style": [
            "error",
            "unix"
        ],
        "quotes": [
            "error",
            "double"
        ],
        "semi": [
            "error",
            "always"
        ]
    }
};
```

*.eslintrc.json:*

```js
{
    "env": {
        "browser": true,
        "commonjs": true,
        "es6": false,
        "node": true
    },
    "parserOptions": {
        "ecmaFeatures": {
            "jsx": true
        },
        "sourceType": "script"
    },
    "rules": {
        "no-const-assign": "warn",
        "no-this-before-super": "warn",
        "no-undef": "warn",
        "no-unreachable": "warn",
        "no-unused-vars": "warn",
        "constructor-super": "warn",
        "valid-typeof": "warn"
    }
}
```

**What did you do? Please include the actual source code causing the issue.**

Here is a small snippet that should be able to be copy-pasted into a new file to replicate the issue:

```js
function runScript(jOrderNum, jShipDate, jInitials, jOverruns) {
    /********** Preprocessor Directives **********/
    //@strict on
    //@target indesign
    //@targetengine "session"
    //@include "logging.jsxinc"

    /********** Output Script v6.50 by Chris McGee **********/
    /****************** Updated 09/20/2016 ******************/
    var windowTitle = "Output Script v6.50 by Chris McGee";
    var finalValue = 31; // The final value of the progress bar window.
    var debug = true; // If set to true, enable all $.writeln()s.
    var printSpecs = false; // Spec sheets will/will not be printed.
    var printOutput = false; // Output pages (Pad, Screen) will/will not be printed.
}

runScript("123456", "2016-10-15", "CM", "true");
```

And, of course, something strange is happening when I paste it here. I will try to attach the full .jsx file where the problem is occurring:

[OutputJ.jsx.zip](https://github.com/eslint/eslint/files/485846/OutputJ.jsx.zip)

**What did you expect to happen?**
I expected that there might be some linting warnings/errors, but not so many at the beginning of the file, and all giving me false information about indenting.

**What actually happened? Please include the actual, raw output from ESLint.**
Every single line that is not a comment is throwing a linting warning that the expected indentation is 1, but it found 0. This is false, as I have made sure there is definitely one tab's worth of spacing (4 spaces) before every one of the lines it is complaining about.

I cannot paste the raw output from ESLint, since it is showing up as a list in the Problems pane of VS Code. With some step-by-step guidance on how to do this from the CLI, I could most likely get what you need.

**Comment by not-an-aardvark** [comment]
> This is false, as I have made sure there is definitely one tab's worth of spacing (4 spaces) before every one of the lines it is complaining about

I think the issue is that your config is trying to find actual *tabs*, not one tab's worth of spacing. It's giving that error message because your files are indented with 0 tabs (and 4 spaces).

In the next version of ESLint, the error message will be a bit clearer (see #7076). In your case it would be something like "Expected indentation of 1 tab but found 4 spaces."

See also: #3638. (Actually, #3638 can probably be closed now that #7076 fixes the error messages.)

**Comment by SturmB** [comment]
Hey, thanks for the quick reply, @not-an-aardvark!

I'm glad to see that such an update regarding indentation is definitely in the works, but that still leaves a question: _Why is ESLint expecting a tab?_ I've already set the file to "Indent Using Spaces" in the Command Palette of VS Code, and it doesn't have these indentation warnings on another file that starts out in the exact same way (attached: [ProofJ.jsx.zip](https://github.com/eslint/eslint/files/487212/ProofJ.jsx.zip)), and uses the same Indent Using Spaces setting. As far as I can tell, ESLint should be expecting 4 spaces, not 1 tab. (I mean, if I have to go back to using tabs for indention, I'll gladly do so, as long as it satisfies the linter.)

**Comment by platinumazure** [comment]
You showed both an .eslintrc and .eslintrc.json. Are those in the same directory?

**Comment by SturmB** [comment]
That is correct. And unless I'm mistaken, the settings in them should apply to all files in subdirectories, too.

**Comment by platinumazure** [comment]
I'm pretty sure ESLint will only read one of those files per directory, but I could be wrong. You should use `eslint --print-config someFile.js` to double check that both config files are being considered.

**Comment by eslintbot** [comment]
Hi @SturmB, thanks for the issue. It looks like there's not enough information for us to know how to help you.

If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

Requesting a rule change? Please see [Proposing a Rule Change](http://eslint.org/docs/developer-guide/contributing/rule-changes) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by SturmB** [comment]
So apparently, I was incorrect. Run on a file in the same folder as those two settings files, I get:

`apaches-iMac:sky-artdept apache$ eslint --print-config OutputJ.jsx
{
  "globals": {},
  "env": {},
  "rules": {
    "no-alert": "off",
    "no-array-constructor": "off",
    "no-bitwise": "off",
    "no-caller": "off",
    "no-case-declarations": "error",
    "no-catch-shadow": "off",
    "no-class-assign": "error",
    "no-cond-assign": "error",
    "no-confusing-arrow": "off",
    "no-console": "error",
    "no-const-assign": "error",
    "no-constant-condition": "error",
    "no-continue": "off",
    "no-control-regex": "error",
    "no-debugger": "error",
    "no-delete-var": "error",
    "no-div-regex": "off",
    "no-dupe-class-members": "error",
    "no-dupe-keys": "error",
    "no-dupe-args": "error",
    "no-duplicate-case": "error",
    "no-duplicate-imports": "off",
    "no-else-return": "off",
    "no-empty": "error",
    "no-empty-character-class": "error",
    "no-empty-function": "off",
    "no-empty-pattern": "error",
    "no-eq-null": "off",
    "no-eval": "off",
    "no-ex-assign": "error",
    "no-extend-native": "off",
    "no-extra-bind": "off",
    "no-extra-boolean-cast": "error",
    "no-extra-label": "off",
    "no-extra-parens": "off",
    "no-extra-semi": "error",
    "no-fallthrough": "error",
    "no-floating-decimal": "off",
    "no-func-assign": "error",
    "no-implicit-coercion": "off",
    "no-implicit-globals": "off",
    "no-implied-eval": "off",
    "no-inline-comments": "off",
    "no-inner-declarations": "error",
    "no-invalid-regexp": "error",
    "no-invalid-this": "off",
    "no-irregular-whitespace": "error",
    "no-iterator": "off",
    "no-label-var": "off",
    "no-labels": "off",
    "no-lone-blocks": "off",
    "no-lonely-if": "off",
    "no-loop-func": "off",
    "no-mixed-requires": "off",
    "no-mixed-spaces-and-tabs": "error",
    "linebreak-style": [
      "error",
      "unix"
    ],
    "no-multi-spaces": "off",
    "no-multi-str": "off",
    "no-multiple-empty-lines": "off",
    "no-native-reassign": "off",
    "no-negated-condition": "off",
    "no-negated-in-lhs": "error",
    "no-nested-ternary": "off",
    "no-new": "off",
    "no-new-func": "off",
    "no-new-object": "off",
    "no-new-require": "off",
    "no-new-symbol": "error",
    "no-new-wrappers": "off",
    "no-obj-calls": "error",
    "no-octal": "error",
    "no-octal-escape": "off",
    "no-param-reassign": "off",
    "no-path-concat": "off",
    "no-plusplus": "off",
    "no-process-env": "off",
    "no-process-exit": "off",
    "no-proto": "off",
    "no-redeclare": "error",
    "no-regex-spaces": "error",
    "no-restricted-globals": "off",
    "no-restricted-imports": "off",
    "no-restricted-modules": "off",
    "no-restricted-syntax": "off",
    "no-return-assign": "off",
    "no-script-url": "off",
    "no-self-assign": "error",
    "no-self-compare": "off",
    "no-sequences": "off",
    "no-shadow": "off",
    "no-shadow-restricted-names": "off",
    "no-whitespace-before-property": "off",
    "no-spaced-func": "off",
    "no-sparse-arrays": "error",
    "no-sync": "off",
    "no-ternary": "off",
    "no-trailing-spaces": "off",
    "no-this-before-super": "error",
    "no-throw-literal": "off",
    "no-undef": "error",
    "no-undef-init": "off",
    "no-undefined": "off",
    "no-unexpected-multiline": "error",
    "no-underscore-dangle": "off",
    "no-unmodified-loop-condition": "off",
    "no-unneeded-ternary": "off",
    "no-unreachable": "error",
    "no-unused-expressions": "off",
    "no-unused-labels": "error",
    "no-unused-vars": "error",
    "no-use-before-define": "off",
    "no-useless-call": "off",
    "no-useless-concat": "off",
    "no-useless-constructor": "off",
    "no-useless-escape": "off",
    "no-void": "off",
    "no-var": "off",
    "no-warning-comments": "off",
    "no-with": "off",
    "no-magic-numbers": "off",
    "array-bracket-spacing": "off",
    "array-callback-return": "off",
    "arrow-body-style": "off",
    "arrow-parens": "off",
    "arrow-spacing": "off",
    "accessor-pairs": "off",
    "block-scoped-var": "off",
    "block-spacing": "off",
    "brace-style": "off",
    "callback-return": "off",
    "camelcase": "off",
    "comma-dangle": "error",
    "comma-spacing": "off",
    "comma-style": "off",
    "complexity": [
      "off",
      11
    ],
    "computed-property-spacing": "off",
    "consistent-return": "off",
    "consistent-this": "off",
    "constructor-super": "error",
    "curly": "off",
    "default-case": "off",
    "dot-location": "off",
    "dot-notation": "off",
    "eol-last": "off",
    "eqeqeq": "off",
    "func-names": "off",
    "func-style": "off",
    "generator-star-spacing": "off",
    "global-require": "off",
    "guard-for-in": "off",
    "handle-callback-err": "off",
    "id-length": "off",
    "indent": [
      "error",
      "tab"
    ],
    "init-declarations": "off",
    "jsx-quotes": "off",
    "key-spacing": "off",
    "keyword-spacing": "off",
    "lines-around-comment": "off",
    "max-depth": "off",
    "max-len": "off",
    "max-nested-callbacks": "off",
    "max-params": "off",
    "max-statements": "off",
    "max-statements-per-line": "off",
    "new-cap": "off",
    "new-parens": "off",
    "newline-after-var": "off",
    "newline-before-return": "off",
    "newline-per-chained-call": "off",
    "object-curly-spacing": [
      "off",
      "never"
    ],
    "object-shorthand": "off",
    "one-var": "off",
    "one-var-declaration-per-line": "off",
    "operator-assignment": "off",
    "operator-linebreak": "off",
    "padded-blocks": "off",
    "prefer-arrow-callback": "off",
    "prefer-const": "off",
    "prefer-reflect": "off",
    "prefer-rest-params": "off",
    "prefer-spread": "off",
    "prefer-template": "off",
    "quote-props": "off",
    "quotes": [
      "error",
      "double"
    ],
    "radix": "off",
    "id-match": "off",
    "id-blacklist": "off",
    "require-jsdoc": "off",
    "require-yield": "off",
    "semi": [
      "error",
      "always"
    ],
    "semi-spacing": "off",
    "sort-vars": "off",
    "sort-imports": "off",
    "space-before-blocks": "off",
    "space-before-function-paren": "off",
    "space-in-parens": "off",
    "space-infix-ops": "off",
    "space-unary-ops": "off",
    "spaced-comment": "off",
    "strict": "off",
    "template-curly-spacing": "off",
    "use-isnan": "error",
    "valid-jsdoc": "off",
    "valid-typeof": "error",
    "vars-on-top": "off",
    "wrap-iife": "off",
    "wrap-regex": "off",
    "yield-star-spacing": "off",
    "yoda": "off"
  },
  "parserOptions": {
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "parser": null,
  "ecmaFeatures": {},
  "extends": "eslint:recommended"
}
apaches-iMac:sky-artdept apache$ 
`

Yet when I go into the Test/ subdirectory and run the same command on a file there (and, yes, this file exists both in the parent and the child, but they are different):

`apaches-iMac:Test apache$ eslint --print-config OutputJ.jsx
{
  "globals": {},
  "env": {},
  "rules": {},
  "parserOptions": {},
  "parser": "espree"
}`

And run from the command line, the version of eslint shows:

`apaches-iMac:Test apache$ eslint -v
v2.7.0`

Not sure if that's the same one used by VS Code, however. the 1.0.6 version I mentioned before is the version of the VS Code plugin.

**Comment by platinumazure** [comment]
@SturmB If you want to do multi-line preformatted text (such as for code blocks or console output), you should use three backticks and then begin your text on a new line. Close also with three backticks on its own line. You can also throw in a language identifier on the first line, such as "js" for JavaScript. I edited your post so you can see how it's done.

**Comment by SturmB** [comment]
That's awesome, @platinumazure! It's totally okay by me, and thank you for letting me know so I can format my posts properly from now on! :~)

**Comment by platinumazure** [comment]
Okay, so I think @not-an-aardvark was correct earlier-- your file seems to have 4 spaces in it, but ESLint's indent rule, when configured with `["error", "tab"]`, requires actual tab characters. If you want to have "1 tab's worth of spaces", then you should use `["error", 4]`. Hope that helps!

I'm not sure why `--print-config` didn't show a whole config for the test file, but I'd say let's start with the config change and then we can re-evaluate from there.

**Comment by SturmB** [comment]
Aha! so making that change to the `.eslintrc.js` file in the root folder of the project seems to have done the trick. And it does, as expected, apply to the files in the Test/ subdirectory, too.

Now I just have to figure out a good way to have it ignore all of the ExtendScript-specific javascript objects and commands. However, I thought that "Folder" was a built-in Javascript object. Perhaps I was wrong.
<img width="551" alt="screen shot 2016-09-22 at 11 42 04 am" src="https://cloud.githubusercontent.com/assets/6216224/18755313/bbcb0ab0-80b9-11e6-9403-499f73ff67b9.png">

Anyway, thanks for the awesome help, @platinumazure!

**Comment by ilyavolodin** [comment]
Closing, as the original question has been answered.

**Comment by JarLowrey** [comment]
When I had this issue in VS Code I had to edit my `.editorconfig` file in the root of the project to match whatever eslint told me to do. 

Example:

```
root = true

[*]
indent_style = tab # change indent style to match warning
indent_size = 2
charset = utf-8
trim_trailing_whitespace = true
insert_final_newline = true
end_of_line = lf
# editorconfig-tools is unable to ignore longs strings or urls
max_line_length = null
```
